orange-tree.rb

```diff
@@ -0,0 +1,132 @@
+# this one is another group project! three adies to a group. :)
+# https://github.com/Ada-Developers-Academy/daily-curriculum/blob/master/topic_resources/orange_tree.md
+# -------------------------------------
+# actually, according to the new dinner requirements, I think we can move on to dessert!
+
+class OrangeTree
+
+  attr_reader :age, :alive, :grove, :height, :orange_count
+
+  def initialize
+    @age = 0
+    @alive = true
+    @height = 0
+    @orange_count = 0
+  end
+
+  def grow_oranges
+    if @age >= 3 # oranges don't grow on young trees!
+      if @grove.soil_quality == "unhealthy"
+        @orange_count = @age # not many if poor soil quality
+
+      elsif @grove.soil_quality == "average"
+        @orange_count = @age * 2 # a few more if average
+
+      else
+        @orange_count = @age * 3 # otherwise healthy amounts!
+      end
+
+    end
+
+  end
+
+  def grow_taller
+    if @age < 5
+      @height += 2
+    else
+      @height += 0.1
+    end
+  end
+
+  def kill_tree
+    @alive = false
+    @orange_count = 0
+    @age = 0 # necessary?
```

Comment:
Not really necessary. Though you raise an interesting point to consider. What happens to your trees that are dead? Do they still grow taller? Looking at your one_year_passes method it looks like you check for this so they don't grow taller, or produce oranges. Looking at your grow_older method it looks like you do not check for this. So resetting the age to zero would mean that the age of a dead tree would be the years that it has been dead. Could be a helpful metric if that is what you are wanting to measure. It all depends on what you want the age of a dead tree to mean.